### DISCUSSION TITLE
user-data-dir session error

### DISCUSSION THREAD

**Discussion by jabacore** [ultrafunkamsterdam/undetected-chromedriver#1101] [Q&A]
Hi,
i am trying to get a real google chrome profile

options = webdriver.ChromeOptions() 
options.add_argument('--user-data-dir=C:/Users/GOD/AppData/Local/Google/Chrome/User Data')
options.add_argument('--profile-directory=Profile 2')

and when starting the browser opens, but it no longer redirects me to the get url, it stays blank. the profile takes it perfectly! I need help thank you very much!

**Comment by strukiii** [comment]
use the constructor parameter user_data_dir where chromeProfile is a file path

`driver = uc.Chrome(
        user_data_dir=chromeProfile,
)`

**Comment by strukiii** [comment]
Also worth noting with chrome profiles, after a while they seem to break everything.  This has been explained in an issue somewhere, not certain what the cause is but if your script starts acting funky delete the profile file and it seems to work fine again.  You'll need to redo whatever config you had that you want to persist between instances.

**Comment by rohit1183** [comment]
How do I specify the profile name? For example, on my mac, this is the profile path when I do chrome://version:
Users/rohit/Library/Application Support/Google/Chrome/Profile 5

I am initializing the driver using this:
uc.Chrome(headless=True, user_data_dir='/Users/rohit/Library/Application\ Support/Google/Chrome/Profile\ 5/')


However, I open the chrome version page using the UC driver, the profile path is listed as:
/Users/rohit/Library/Application\ Support/Google/Chrome/Profile\ 5\Default

How do I make sure that Default is not appended at the end and that the driver actually uses Profile 5? In the vanilla version, we could specify both the profile path and the profile name. Here I am only seeing the option to provide the path.

**Reply by strukiii** [reply]
I'm on windows, I name the directory with the file path `'d:\\Temp\\profile-0'`

**Comment by orhrecords** [comment]
try this way
options = webdriver.ChromeOptions()
options.add_argument('--user-data-dir=C:\\Users\\GOD\\AppData\\Local\\Google\\Chrome\\User Data\\')
options.add_argument('--profile-directory=Profile 2'

**Comment by orhrecords** [comment]
No. It not going to work like that

You of to have --user-dir and the profile separate with the two backword
slash

**Comment by Anticope12** [comment]
try to to make sure you force close all the chrome operation before running your code
